### ISSUE TITLE
Video playing behind preroll ad

### ISSUE THREAD

**Issue by davidjamesherzog** [googleads/videojs-ima#663]
I have run into an issue where the video starts playing at the same time as the preroll ad.  So if the preroll ad is 10 seconds long, once the video appears it is already 10 seconds into the video.  I've created a codepen below to demonstrate code similar to what we are using in production.  This example works just fine.

https://codepen.io/davidjamesherzog/pen/LrzWZG

Unfortunately, when I put this similar type code in our production code, we run into the issue of the preroll and video playing at the same time.  Our production code has the equivalent code above but is intertwined with a whole bunch of API calls, promises, callbacks, components, etc. which may be causing the problem.  Is there a certain sequence that needs to occur to make sure the ads prevent the video from starting once it is loaded and configured with autoplay on?  I call `video.ima.requestAds();` prior to loading the source into the player and that seems to work just fine in the sample above but does not work in our production code.

**Comment by shawnbuso** [comment]
Hey - I think this sample is the same or very similar to the one provided in #662. It's erroring out for me because it's trying to load controller.js, which doesn't exist. Since you said this example is working but you're seeing issues in production, would it be possible for you to share your production page with us so we can see the issue? Thanks!

**Comment by davidjamesherzog** [comment]
I removed the reference to controller.js. That was a copy/paste error.

**Comment by davidjamesherzog** [comment]
Unfortunately, I can't share the production page.  I have actually taken our custom video code (setting up the player, loading video and listening for events) and commented out almost everything and only left the bare essentials.  The issue goes away so it is somewhere in our code.  I started to uncomment out the code piece by piece but I got pulled off onto other priorities.  I won't get back to looking at this for another week or two.

**Comment by shawnbuso** [comment]
OK let me know if you need anything more from us - thanks!

**Comment by davidjamesherzog** [comment]
I think I found out why this was occurring.  I was explicitly setting `autoplay(true)` on the player because of autoplay block detection in the browsers.  When I removed this line, it stopped the video from playing while the preroll was playing.  I'm going to use the can autoplay library that is used in the autoplay example in this repo.  That seems like a much better solution for autoplay detection than my hack.
